### ISSUE TITLE
Rename IPromise.Transform to IPromise.Then

### ISSUE THREAD

**Issue by cgarciae** [Real-Serious-Games/C-Sharp-Promise#4]
Its not ambiguous and you only have to remember 1 method. I use Dart which has a standard Futures library and I find it a lot easier.

**Comment by ashleydavis** [comment]
I wish that were possible. I haven't found a way to have Transform be Then and still compile.

If you can find a way to do this please fork the repo, make the change and submit a pull request.

I'd be very keen to have it work the way you suggest.

**Comment by cgarciae** [comment]
Hi @ashleydavis,

I think #1 made possible refactoring `Transform` to `Then`. I did it on my version project and it worked with no compiler errors out of the box. I'll try to open the git repo by itself and refactor + pass some changes I made:

1. I added a new `public Promise<PromisedT> (PromisedT value)` constructor to create a resolved promise, it seemed important because this is the monadic "Return" function.
2. Changed the internals of `Transform` (now called Then) to just "fallback" to the "bind" `Then` using the new constructor like this: 

```c#
        public IPromise<ConvertedT> Then<ConvertedT>(Func<PromisedT, ConvertedT> f)
        {
            //Argument.NotNull(() => transform);
            return Then((PromisedT t) => (IPromise<ConvertedT>)new Promise<ConvertedT>(f(t)));
        }
```

I have experience with Futures/Promises since I made my own Futures library for the project, as far as I saw it worked perfectly but I decided not to use your library once #1 was solved.
